ai2thor/controller.py

```diff
@@ -1049,73 +1056,105 @@ def _branch_commits(self, branch):
         if branch:
             commits = self._branch_commits(branch)
         elif commit_id:
-            ver_build = ai2thor.build.Build(
-                arch, commit_id, self.include_private_scenes, self.releases_dir
-            )
-
-            if not ver_build.exists():
-                raise ValueError(
-                    "Invalid commit_id: %s - no build exists for arch=%s"
-                    % (commit_id, arch)
-                )
-
-            return ver_build
+            commits = [commit_id]
         else:
-            git_dir = os.path.normpath(
-                os.path.dirname(os.path.realpath(__file__)) + "/../.git"
-            )
-            commits = (
-                subprocess.check_output(
-                    "git --git-dir=" + git_dir + " log -n 10 --format=%H", shell=True
-                )
-                .decode("ascii")
-                .strip()
-                .split("\n")
-            )
-
-        rdir = self.releases_dir
-        found_build = None
+            commits = self.local_commits()
 
         if local_build:
-            rdir = os.path.normpath(
+
+            releases_dir = os.path.normpath(
                 os.path.dirname(os.path.realpath(__file__)) + "/../unity/builds"
             )
             commits = [
-                "local"
+                ai2thor.build.LOCAL_BUILD_COMMIT_ID
             ] + commits  # we add the commits to the list to allow the ci_build to succeed
 
-        for commit_id in commits:
-            commit_build = ai2thor.build.Build(
-                arch, commit_id, self.include_private_scenes, rdir
-            )
-
-            try:
-                if os.path.isdir(commit_build.base_dir) or (
-                    not local_build and commit_build.exists()
-                ):
-                    found_build = commit_build
-                    break
-            except Exception:
-                pass
+        request = ai2thor.platform.Request(
+            platform.system(), self.width, self.height, self.x_display, self.headless
+        )
+        builds = self.find_platform_builds(request, commits, releases_dir, local_build)
+        if not builds:
+            if commit_id:
+                raise ValueError(
+                    "Invalid commit_id: %s - no build exists for arch=%s"
+                    % (commit_id, platform.system())
+                )
+            else:
+                raise Exception(
+                    "No build exists for arch=%s and commits: %s"
+                    % (platform.system(), ", ".join(map(lambda x: x[:8], commits)))
+                )
 
-        if commit_build and commit_build.commit_id != commits[0]:
-            warnings.warn(
-                "Build for the most recent commit: %s is not available.  Using commit build %s"
-                % (commits[0], commit_build.commit_id)
+        # select the first build + platform that succeeds
+        # For Linux, this will select CloudRendering, then Linux64 (X11)
```

Comment:
Currently we don't select cloud rendering first right? As it's still being validated?

Reply:
I updated the comment to reflect the new behavior. CloudRendering is disabled by default.